Zip file extraction

miriamkw opened the pull request:
Summary of PR:
- Each study can now eigther be processed directly from the .zip file, or the extracted folders. This is automatically detected
    - Note that the processing will directly read from within the .zip without extracting them, with exception of Loop study that gets extracted before using the dask solution
- Flair and Repliace BG scripts did not work because the processing is written for if you have the entire old version of the dataset. This was handled by adding a mock dataframe for the unavailable data, but please let me know how you would like to handle that approperiately

miriamkw committed:
Working example of extracting directly t1dezi from the zip file

miriamkw committed:
Updated readme about whether to unzip or not

miriamkw committed:
cleaned up code to reduce redundancy t1dexi parser

miriamkw committed:
Adding zip file extraction / inspection for all studies

miriamkw committed:
Merge branch 'develop' of github.com:nudgebg/egvinsulin into zip_file_extraction

miriamkw committed:
Smallfix PEDAP

janvv commented:
Excellent. I will review the PR this week.

janvv commented on src/pandas_helper.py:
Please add docstrings.

janvv commented on studies/flair.py:
i think it would be better here to check if the relevant files exist (and throw an error if not) instead of working with mock data. We might want to batch process all study datasets in try blocks.

janvv commented on studies/loop.py:
```diff
@@ -2,9 +2,19 @@
 
 from .studydataset import StudyDataset
 
+def unzip_folder(zip_path, extract_to):
+    """Unzips a folder given its path."""
+    #extract_to = os.path.dirname(zip_path)  # Extract in the same directory
+
+    with zipfile_deflate64.ZipFile(zip_path, 'r') as zip_ref:
+        zip_ref.extractall(extract_to)
+        print(f"Extracted to: {extract_to}")
+
+
```
use the Logger instead of print statements, remove commented lines and use google styled docstring format.

janvv commented on studies/replacebg.py:
```diff
@@ -6,6 +6,13 @@
 import os
 from src import pandas_helper, logger
 
+def get_basal_df_mock():
+    df = pd.DataFrame(columns=['PtID', 'datetime', 'DeviceDtTmDaysFromEnroll', 'DeviceTm', 'Duration',
+                          'ExpectedDuration', 'SuprDuration', 'RecID', 'Rate'])
+    df['Rate'] = pd.to_numeric(df['Rate'])
+    return df
+
+
```
same as before, let's throw an error when load_data() fails.

janvv reviewed (changes requested):
Excellent additions @miriamkw! Executed without problems on my end.

I left comments in the files.  

Before merging the PR i have some change requests:
 1. Please use Google styled docstrings
 2. The use of empty mock dataframes is a great workaround but i don't think we should support it this way. We can't really support all datasets in all their release versions. For now i would suggest the function should throw an error when trying to access the missing file (even might hint the user that data is missing). But this is out of scope for this PR. Let's make sure you have access to all the data.

janvv commented on studies/replacebg.py:
```diff
@@ -60,7 +71,7 @@ def _load_data(self, subset: bool = False):
         
         #drop patients that are not in all datasets 
-        patient_ids_to_keep = reduce(np.intersect1d, [df_basal['PtID'].unique(), 
+        patient_ids_to_keep = reduce(np.intersect1d, [#df_basal['PtID'].unique(),
```
This change affects also complete datasets which shouldn't happen

miriamkw committed:
Added docstrings to pandas_helpers

miriamkw committed:
cleaning up comments/prints and adding doc strings

miriamkw committed:
Reverted basal mock temporary solutions

miriamkw committed:
Merge branch 'develop' into zip_file_extraction

miriamkw committed:
Removed .zip from output folder names so its conistent between zip / not zip

miriamkw committed:
Merge remote-tracking branch 'refs/remotes/origin/zip_file_extraction' into zip_file_extraction

miriamkw commented:
> Excellent additions @miriamkw! Executed without problems on my end.
> 
> I left comments in the files.
> 
> Before merging the PR i have some change requests:
> 
> 1. Please use Google styled docstrings
> 2. The use of empty mock dataframes is a great workaround but i don't think we should support it this way. We can't really support all datasets in all their release versions. For now i would suggest the function should throw an error when trying to access the missing file (even might hint the user that data is missing). But this is out of scope for this PR. Let's make sure you have access to all the data.

Thank you for the comments. I have now made the revisions addressing the comments, and reverted the changes in Failr and Replace BG so that the code throws an error when insulin data is not available. 
A separate smallfix was added to ensure that the output folders never have ".zip" in them. 
I reran the code on the data you gave me, and everything looks fine on my end.
